### PR TITLE
More test harnessing of the usage statistics structure

### PR DESCRIPTION
**Pull request by batmat**
- added an agent so that the test is closer to reality
- tested every fields (compared to only a subset previously)

Cherry-picked and squashed the related modifications from #1985 as that PR is probably less likely to get merged soon and I think there's some value to harness the current usage stats structure.

### CODE REVIEW

**Comment by batmat** [comment]
@daniel-beck that one, as a subset of #1985, should be more acceptable in the short term I guess, as it improves the usage statistics coverage (though doesn't test the opt-in/opt-out behaviour).

**Comment by batmat** [comment]
ping @jenkinsci/code-reviewers (note: that PR only touches test code)

File: test/src/test/java/hudson/model/UsageStatisticsTest.java [hunk 1]
```diff
@@ -68,7 +83,76 @@ public void roundtrip() throws Exception {
...
+
+    // Plugins can be retrieved in any order, so sorting them so that the test is stable
+    private Object sortPlugins(List<JSONObject> list) {
+        List<JSONObject> sorted = new ArrayList<>(list);
+        Collections.sort(sorted, new Comparator<JSONObject>() {
+            public int compare(JSONObject j1, JSONObject j2) {
+                return j1.getString("name").compareTo(j2.getString("name"));
+            }
+
+            ;
```
Comment (oleg-nenashev) [inline test/src/test/java/hudson/model/UsageStatisticsTest.java]: NIT: strange formatting glitches

**Comment by oleg-nenashev** [comment]
I would move the resources to the UsageStatisticsTest package

**Comment by batmat** [comment]
OK, missed the formatting glitch 'cause of the default IJ folding and
didn't catch it while staging it.
Also agree with moving the .json files to a dedicated package.
Will do all this later this day.
Thanks for the review Oleg.

2016-04-08 9:44 GMT+00:00 Oleg Nenashev <notifications@github.com>:

**Comment by batmat** [comment]
@oleg-nenashev FYI just force pushed a fixed version of the commit.

**Comment by oleg-nenashev** [comment]
:+1:

**Pull request merged**